chat-app/src/MessageForm.js

```diff
@@ -0,0 +1,47 @@
+// import 'dotenv/config';
+import { useState } from 'react';
+
+export const MessageForm = ({ messageText, setMessageText, sendMessage }) => {
+  const [loading, setLoading] = useState(false);
+  const [error, setError] = useState(null);
+
+  const handleSubmit = async (event) => {
+    event.preventDefault();
+    setLoading(true);
+    setError(null);
+
+    try {
+      await sendMessage();
+    } catch (err) {
+      setError("Failed to send message");
+      console.error(err);
+    } finally {
+      setLoading(false);
+    }
+  };
+
+  return (
+    <form
+    className="message-form"
+    id="message-form"
+    onSubmit={handleSubmit}
+    >
+
+    <input
+      type="text"
+      name="message"
+      id="message-input"
+      className="message-input"
+      value={messageText}
+      onChange={(event) => setMessageText(event.target.value)}
+      placeholder='Type your message...'
+      disabled = {loading}
```

Comment:
Disabling the input field while loading is a good practice to prevent duplicate submissions. Ensure that the user is informed about the loading state, possibly by changing the button text to 'Sending...' or showing a spinner.